doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+
+We'll need to introduce a new keybinding command that should indicate that the
+key is unbound. We'll load the user keybindings and layer them on the defaults
+as described above. If during the deserializing we find an entry that's bound to
+the command `"unbound"` or any other string that we don't understand, instead of
+trying to _set_ the keybinding, we'll _clear_ the keybinding with a new method
+`AppKeyBindings::ClearKeyBinding(chord)`.
+
+### Removing the Globals Object
+
+As a part of #[1005](https://github.com/microsoft/terminal/pull/1005), all the
+global settings were moved to their own object within the serialized settings.
+This was to try and make the file easier to parse as a user, considering global
+settings would be intermingled with profiles, keybindings, color schemes, etc.
+Since this change will make the user settings dramatically easier to navigate,
+we should probably remove the `globals` object, and have globals at the root
+level again.
+
+### Default `profiles.json`
+
+Below is an example of what the default user settings file might look like when
+it's first generated, taking all the above points into consideration.
+
+```js
+// To view the default settings, open <path-to-app-package>\defaults.json
+{
+    "defaultProfile" : "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
+    "profiles": [
+        {
+            // Make changes here to the cmd.exe profile
+            "guid": "{6239a42c-1de4-49a3-80bd-e8fdd045185c}"
+        },
+        {
+            // Make changes here to the Windows Powershell profile
+            "guid": "{086a83cd-e4ef-418b-89b1-3f6523ff9195}",
+        },
+        {
+            "guid": "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
+            "name" : "Powershell Core",
+            "commandline" : "pwsh.exe",
+            "icon" : "ms-appx:///ProfileIcons/{574e775e-4f2a-5b96-ac1e-a2962a402336}.png",
+        }
+    ],
+
+    // Add custom color schemes to this array
+    "schemes": [],
+
+    // Add any keybinding overrides to this array.
+    // To unbind a default keybinding, set the command to "unbound"
+    "keybindings": []
+}
+
+```
+
+Note the following:
+* cmd.exe and powershell.exe are both in the file, as to give users an easy
+  point to extend the settings for those default profiles.
+* Powershell Core is included in the file, and the default profile has been set
+  to its GUID.
+* There are a few helpful comments scattered throughout the file to help point
+  the user in the right direction.
+
+### Re-ordering profiles
+
+Since there are shortcuts to open the Nth profile in the list of profiles, we
+need to expose a way for the user to change the order of the profiles. This was
+not a problem when there was only a single list of profiles, but if the defaults
+are applied _first_ to the list of profiles, then the user wouldn't be able to
+change the order of the default profiles. Additionally, any profiles they add
+would _always_ show up after the defaults.

```

Comment:
So then the ordering of the profiles will always be [ {default}, {custom} ]? Where do dynamic profiles fit in? I'd like it if I didn't have to check the ordering for each machine I have; that way, I would always know what ALT+8 is mapped to.

For example, I'm on machine 1 with the list { cmd, pwsh, ubuntu-custom, azure}. Then I move to machine 2 where ubuntu-custom doesn't exist. Could I still do ALT+3, and just get a dialogue box saying "ubuntu-custom isn't available on your machine"? That way, I'll always know that ALT+4 will launch azure on both machines?

Reply:
the ordering of the profiles would always be [ { things the user knew about & existed }, { things that are newly-generated } ]

For "constant keybindings", i think we should instead allow binding keys to profiles directly (instead of profile numbers)... #1349 !

Reply:
No, the ordering will be whatever order your user settings order them in, followed by the rest of them. So I guess [{user profiles}, {defaults - the ones in the user profiles}, {dynamics - ones from user profiles}].

I don't currently have any provision for that, though I guess that's a good point that the keybindings will affect different profiles. I'm not sure how much that'll matter in the wake of #1770. Much of this spec was written before we decided that roaming was bad, so some of it won't apply anymore.

Reply:
(github didn't reload between the time Dustin posted his comment and I posted mine - I was replying to Carlos, not Dustin ;__;)